java-frontend/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -72,9 +81,21 @@
     "com.sun.org.apache.xml.internal.security.utils.UnsyncByteArrayOutputStream",
     "org.springframework.context.ConfigurableApplicationContext"
   };
+  
+  private static final List<String> excludedTypesList = new ArrayList<>();
+  
   private static final MethodMatcher[] CLOSEABLE_EXCEPTIONS = new MethodMatcher[] {
     MethodMatcher.create().typeDefinition("java.nio.file.FileSystems").name("getDefault").withoutParameter()
   };
+  
+  @Override
+  public void init(MethodTree methodTree, CFG cfg) {
```

Comment:
Please do not override the `init()` method. Symbolic execution based checks are sensitive, and in this case we don't want to touch anything in the initialization phase.

The best approach in order to retrieve the list of excluded resource types is to use a dedicated (private) getter, which lazily initialize the list (which then should not be static, with all the side effects you noticed).